app/screens/edit_profile/components/custom_profile_field.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useRef} from 'react';
+import {type TextInputProps} from 'react-native';
+
+import {type FloatingTextInputRef} from '@components/floating_text_input_label';
+
+import Field from './field';
+
+export type FieldProps = TextInputProps & {
+    isDisabled?: boolean;
+    fieldKey: string;
+    label: string;
+    maxLength?: number;
+    onTextChange: (fieldKey: string, value: string) => void;
+    isOptional?: boolean;
+    testID: string;
+    error?: string;
+    value: string;
+    onFocusNextField: (fieldKey: string) => void;
+};
+
+const CustomProfileField = ({
+    autoCapitalize = 'none',
+    autoCorrect = false,
+    fieldKey,
+    isDisabled = false,
+    isOptional = false,
+    keyboardType = 'default',
+    label,
+    maxLength,
+    onTextChange,
+    testID,
+    value,
+    error,
+    onFocusNextField,
+    ...props
+}: FieldProps) => {
+    const ref = useRef<FloatingTextInputRef>(null);
```

Comment:
this is the reason for this new component, I couldn't find a way to create refs in runtime